### PR TITLE
add bdd test on nebula cluster graceful termination

### PR DESCRIPTION
**Pull request by kikimo**
this pr add bdd integration test on issue #3005 #3122 and pr #3014

### CODE REVIEW

File: tests/cluster/nebula_cluster.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/python3
```
Comment (HarrisChu) [inline tests/cluster/nebula_cluster.py:1]: please add the copyright

File: tests/cluster/nebula_cluster.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    def check_cluster_stopped(self):
+        return check_process_stopped(self.pids_dir)
+
+    def prepare(self):
+        Path(self.data_dir).mkdir(parents=True, exist_ok=True)
+        Path(self.etc_dir).mkdir(parents=True, exist_ok=True)
+        Path(self.logs_dir).mkdir(parents=True, exist_ok=True)
+        Path(self.pids_dir).mkdir(parents=True, exist_ok=True)
+
+        default_meta_conf_file = os.path.join(self.default_conf_dir, 'nebula-metad.conf.production')
```
Comment (HarrisChu) [inline tests/cluster/nebula_cluster.py:129]: copy the configs into another folder?

**Comment by HarrisChu** [comment]
generally LGTM.

additional two points:
1. how about we use nebula cluster by default.
2. more flexible, so we can test zone in future. e.g. `Given a nebulacluster with 1 graphd and 1 metad and 6 storaged`

I would follow these two.

File: tests/tck/cluster/terminate.feature [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@cluster
+Feature: Nebula service termination test
+  All nebula services shold exit as expected after termination
+
+  Scenario: Basic termination test
+    Given a small nebula cluster
+    When the cluster was terminated
+    Then no service should still running after 4s
```
Comment (Aiee) [inline tests/tck/cluster/terminate.feature:8]: Please add the newline at the end of the file

File: tests/tck/cluster/terminate.feature [hunk 4]
```diff
@@ -0,0 +1,8 @@
+@cluster
```
Comment (CPWstatic) [inline tests/tck/cluster/terminate.feature:1]: This annotation should deleted after all tests passed.

**Review by CPWstatic — COMMENTED** [review]
LGTM.

**Comment by kikimo** [comment]
closed, @HarrisChu create a new pr

**Pull request closed without merging**